src/ansys/meshing/prime/graphics/plotter.py

```diff
@@ -0,0 +1,353 @@
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""Module for the plotter."""
+import enum
+import warnings
+from typing import Any, Dict, List, Optional
+
+import numpy as np
+from ansys.tools.visualization_interface import Plotter
+from ansys.tools.visualization_interface.backends.pyvista import PyVistaBackend
+
+import ansys.meshing.prime as prime
+from ansys.meshing.prime.core.mesh import DisplayMeshInfo
+from ansys.meshing.prime.core.model import Model
+from ansys.meshing.prime.graphics.widgets.color_by_type import ColorByTypeWidget
+from ansys.meshing.prime.graphics.widgets.hide_picked import HidePicked
+from ansys.meshing.prime.graphics.widgets.picked_info import PickedInfo
+from ansys.meshing.prime.graphics.widgets.toggle_edges import ToggleEdges
+
+color_matrix = np.array(
+    [
+        [155, 186, 126],
+        [242, 236, 175],
+        [255, 187, 131],
+        [194, 187, 97],
+        [159, 131, 169],
+        [157, 190, 139],
+        [233, 218, 158],
+        [254, 252, 196],
+        [246, 210, 148],
+        [215, 208, 198],
+        [196, 235, 145],
+    ]
+)
+
+
+class ColorByType(enum.IntEnum):
+    """Contains the zone types to display."""
+
+    ZONE = 0
+    ZONELET = 1
+    PART = 2
+
+
+class PrimePlotter(Plotter):
+    """Create a plotter for PyPrimeMesh models.
+
+    This plotter is a wrapper around the PyAnsys generic plotter
+    with additional functionality for PyPrimeMesh.
+
+    Parameters
+    ----------
+    use_trame : Optional[bool], optional.
+        Whether to use trame or not, by default None.
+    allow_picking : Optional[bool], optional.
+        Whether to allow picking or not, by default True.
+    """
+
+    def __init__(
+        self, use_trame: Optional[bool] = None, allow_picking: Optional[bool] = True
+    ) -> None:
+        """Initialize the widget."""
+        self._backend = PyVistaBackend(use_trame=use_trame, allow_picking=allow_picking)
+        super().__init__(backend=self._backend)
+
+        # info of the actor to pass to picked info widget
+        self._info_actor_map = {}
+        self._backend.add_widget(ToggleEdges(self))
+        self._backend.add_widget(ColorByTypeWidget(self))
+        self._backend.add_widget(HidePicked(self))
+        self._backend.add_widget(PickedInfo(self))
+
+    @property
+    def info_actor_map(self) -> Dict:
+        """Get the info actor map for picked info widget.
```

Comment:
```suggestion
        """Get the info actor map for the selected information widget.
```
We should spell out words in descriptions.